source/auth.js

```diff
@@ -1,13 +1,80 @@
 "use strict";
 
-function generateBasicAuthHeader(username, password) {
+const crypto = require("crypto");
+
+let username, password;
+
+function generateBasicAuthHeader(user, pass) {
+    username = user;
+    password = pass;
     return "Basic " + Buffer.from(username + ":" + password).toString("base64");
 }
+
 function generateTokenAuthHeader(tokenInfo) {
     return `${tokenInfo.token_type} ${tokenInfo.access_token}`;
 }
 
+function generateDigestAuthHeader(url, options, digest) {
+    function md5(data) {
+        return crypto
+            .createHash("md5")
+            .update(data)
+            .digest("hex");
+    }
+
+    function ha1Compute(algorithm, user, realm, pass, nonce, cnonce) {
+        const ha1 = md5(`${user}:${realm}:${pass}`);
+        if (algorithm && algorithm.toLowerCase() === "md5-sess") {
+            return md5(`${ha1}:${nonce}:${cnonce}`);
+        } else {
+            return ha1;
+        }
+    }
+
+    const _url = url.replace("//", "");
+    const uri = _url.indexOf("/") == -1 ? "/" : _url.slice(_url.indexOf("/"));
+
+    const method = options.method ? options.method.toUpperCase() : "GET";
+
+    const qop = /(^|,)\s*auth\s*($|,)/.test(digest.qop) && "auth";
```

Comment:
Could this be simplified to `\bauth\b`, for instance?

Also I'd prefer it if the value was explicit here.. like `? "auth" : null;` for instance..

Reply:
See https://github.com/perry-mitchell/webdav-client/pull/152/commits/8c52e4ffd8d7265abcc4f01c6e956ef861cb80c8#diff-1995eac40392e12643bbf44abad3cf8cR35

I kept the bulk of the regular expression, though, because I'm not its author and I am not sure if that is handling edge cases.